File: java/src/main/java/org/training/billing/BillingService.java
package org.training.billing;

import com.ecorp.billing.GoodOldBillingSystem;

public class BillingService {
    private GoodOldBillingSystem billingSystem;

    public BillingService(GoodOldBillingSystem billingSystem) {
        this.billingSystem = billingSystem;
    }

    public void invoice(Invoice invoice) {
        int amount = invoice.amount;
        Currency currency = invoice.currency;
        String companyName = invoice.companyName;
        String invoiceForBillingSystem = String.format("%d%s to %s", amount, currency.symbol, companyName);
        billingSystem.invoice(invoiceForBillingSystem);
    }
}

File: kotlin/src/main/kotlin/org/training/billing/BillingService.kt
package org.training.billing

import com.ecorp.billing.GoodOldBillingSystem

class BillingService(private val billingSystem: GoodOldBillingSystem) {

    fun invoice(invoice: Invoice) {
        val invoiceForBillingSystem =
            String.format("%d%s to %s", invoice.amount, invoice.currency.symbol, invoice.companyName)
        billingSystem.invoice(invoiceForBillingSystem)
    }
}
